frontend/test/resolution/testDomains.cpp

```diff
@@ -139,199 +152,268 @@ module M {
-  findVarType(m, rr, "d").dump();
-  findVarType(m, rr, "t").dump();
-  findVarType(m, rr, "i").dump();
 
-  assert(findVarType(m, rr, "equal").isParamTrue());
+  const Variable* d = m->stmt(0)->toVariable();
+  assert(d);
+  assert(d->name() == "d");
+  QualifiedType dType = rr.byAst(d).type();
+  assert(dType.type()->isErroneousType());
+
+  assert(guard.errors().size() == 1);
+  auto& e = guard.errors()[0];
+  assert(e->type() == chpl::InvalidDomainCall);
+
+  printf("Success: cannot resolve incorrect domain type expression %s\n",
+         domainType.c_str());
+
+  guard.clearErrors();
+}
+
+// Ensure we gracefully error for bad domain type expressions, with or without
+// the standard modules available.
+static void testBadDomain(std::string domainType) {
+  // Without standard modules
+  {
+    Context ctx;
+    Context* context = &ctx;
+    ErrorGuard guard(context);
+
+    testBadDomainHelper(domainType, context, guard);
+  }
+
+  // With standard modules
+  {
+    Context::Configuration config;
+    config.chplHome = getenv("CHPL_HOME");
+    Context ctx(config);
+    Context* context = &ctx;
+    setupModuleSearchPaths(context, false, false, {}, {});
+    ErrorGuard guard(context);
+
+    testBadDomainHelper(domainType, context, guard);
+  }
 }
 
 int main() {
-  testRectangular("domain(1)", 1, "int", false);
-  testRectangular("domain(2)", 2, "int", false);
-  testRectangular("domain(1, stridable=true)", 1, "int", true);
-  testRectangular("domain(2, int(8))", 2, "int(8)", false);
-  testRectangular("domain(3, int(16), true)", 3, "int(16)", true);
-  testRectangular("domain(stridable=false, idxType=int, rank=1)", 1, "int", false);
-
-  testAssociative("domain(int)", "int", true);
-  testAssociative("domain(int, false)", "int", false);
-  testAssociative("domain(string)", "string", true);
-
-  testBadPass("domain(1)", "domain(2)");
-  testBadPass("domain(int)", "domain(string)");
-  testBadPass("domain(1)", "domain(int)");
-
-  testIndex("domain(1)", "int");
-  testIndex("domain(2)", "2*int");
-  testIndex("domain(1, bool)", "bool");
-  testIndex("domain(2, bool)", "2*bool");
-  testIndex("domain(int)", "int");
-  testIndex("domain(string)", "string");
+  testRectangular("domain(1)", 1, "int", "one");
+  testRectangular("domain(2)", 2, "int", "one");
+  testRectangular("domain(1, strides=strideKind.one)", 1, "int", "one");
+  testRectangular("domain(2, int(8))", 2, "int(8)", "one");
+  testRectangular("domain(3, int(16), strideKind.negOne)", 3, "int(16)", "negOne");
+  testRectangular("domain(strides=strideKind.negative, idxType=int, rank=1)", 1, "int", "negative");
+
+  // TODO: re-enable associative
+  // testAssociative("domain(int)", "int", true);
+  // testAssociative("domain(int, false)", "int", false);
+  // testAssociative("domain(string)", "string", true);
+
+  // TODO: re-enable
+  // testBadPass("domain(1)", "domain(2)");
+  // testBadPass("domain(int)", "domain(string)");
+  // testBadPass("domain(1)", "domain(int)");
```

Comment:
Do you need to re-enable these as part of this PR? If not, could you note in future work that you've also disabled this functionality?

Reply:
Yes, thanks. Re-enabled these and `testIndex` as well, except for the associative domain cases.